cartography/intel/okta/oktaintel.py

```diff
@@ -0,0 +1,1225 @@
+    usergroups_client = UserGroupsClient(
+        base_url=f"https://{okta_org}.okta.com/",
+        api_token=OKTA_API_KEY,
+    )
+
+    return usergroups_client
+
+
+def _create_application_client(okta_org):
+    """
+    Create Okta AppInstanceClient
+    :param okta_org: Okta organization name
+    :return: Instance of AppInstanceClient
+    """
+    app_client = AppInstanceClient(
+        base_url=f"https://{okta_org}.okta.com/",
+        api_token=OKTA_API_KEY,
+    )
+
+    return app_client
+
+
+def _create_factor_client(okta_org):
+    """
+    Create Okta FactorsClient
+    :param okta_org: Okta organization name
+    :return: Instance of FactorsClient
+    """
+
+    # https://github.com/okta/okta-sdk-python/blob/master/okta/FactorsClient.py
+    factor_client = FactorsClient(
+        base_url=f"https://{okta_org}.okta.com/",
+        api_token=OKTA_API_KEY,
+    )
+
+    return factor_client
+
+
+def _create_api_client(okta_org, path_name):
+    """
+    Create Okta ApiClient
+    :param okta_org: Okta organization name
+    :param path_name: API Path
+    :return: Instance of ApiClient
+    """
+    api_client = ApiClient(
+        base_url=f"https://{okta_org}.okta.com/",
+        pathname=path_name,
+        api_token=OKTA_API_KEY,
+    )
+
+    return api_client
+
+
+def _get_okta_users(user_client):
+    """
+    Get Okta users from Okta server
+    :param user_client: user client
+    :return: Array of dictionary containing user properties
+    """
+    user_list = []
+    paged_users = user_client.get_paged_users()
+    while True:
+        for current_user in paged_users.result:
+            user_props = transform_okta_user(current_user)
+            user_list.append(user_props)
+        if not paged_users.is_last_page():
+            # Keep on fetching pages of users until the last page
+            paged_users = user_client.get_paged_users(url=paged_users.next_url)
+        else:
+            break
+
+    return user_list
+
+
+def transform_okta_user(okta_user):
+    """
+    Transform okta user data
+    :param okta_user: okta user object
+    :return: Dictionary container user properties for ingestion
+    """
+
+    # https://github.com/okta/okta-sdk-python/blob/master/okta/models/user/User.py
+    user_props = {}
+    user_props["first_name"] = okta_user.profile.firstName
+    user_props["last_name"] = okta_user.profile.lastName
+    user_props["login"] = okta_user.profile.login
+    user_props["email"] = okta_user.profile.email
+    user_props["second_email"] = okta_user.profile.secondEmail
+    user_props["mobile_phone"] = okta_user.profile.mobilePhone
+
+    # https://github.com/okta/okta-sdk-python/blob/master/okta/models/user/User.py
+    user_props["id"] = okta_user.id
+    user_props["created"] = okta_user.created.strftime("%m/%d/%Y, %H:%M:%S")
+    if okta_user.activated:
+        user_props["activated"] = okta_user.activated.strftime("%m/%d/%Y, %H:%M:%S")
+    else:
+        user_props["activated"] = None
```

Comment:
I don't think explicitly setting this to None is necessary

Reply:
This is to make sure the property is set and avoids checking in the future and have it in the graph object